app/src/main/java/org/schabi/newpipe/player/Player.java

```diff
@@ -2516,50 +2516,75 @@ public void onPlayerError(@NonNull final ExoPlaybackException error) {
         Log.e(TAG, "ExoPlayer - onPlayerError() called with:", error);
 
         saveStreamProgressState();
-
-        // create error notification
-        final ErrorInfo errorInfo;
-        if (currentMetadata == null) {
-            errorInfo = new ErrorInfo(error, UserAction.PLAY_STREAM,
-                    "Player error[type=" + error.type + "] occurred, currentMetadata is null");
-        } else {
-            errorInfo = new ErrorInfo(error, UserAction.PLAY_STREAM,
-                    "Player error[type=" + error.type + "] occurred while playing "
-                            + currentMetadata.getMetadata().getUrl(),
-                    currentMetadata.getMetadata());
-        }
-        ErrorUtil.createNotification(context, errorInfo);
+        boolean isBehindLiveWindowException = false;
```

Comment:
I think the following would be easier to understand:

```Java
        boolean couldErrorBeFixed = false;

        switch (error.type) {
            case ExoPlaybackException.TYPE_SOURCE:
                couldErrorBeFixed = processSourceError(error.getSourceException());
                break;
            case ExoPlaybackException.TYPE_UNEXPECTED:
                setRecovery();
                reloadPlayQueueManager();
                break;
            case ExoPlaybackException.TYPE_REMOTE:
            case ExoPlaybackException.TYPE_RENDERER:
            default:
                onPlaybackShutdown();
                break;
        }

        if (couldErrorBeFixed) {
            return;
        }

        createErrorNotification(error); // In theory we also don't need this method anymore - you might also remove it

        if (fragmentListener != null) {
            fragmentListener.onPlayerError(error);
        }
```

Reply:
Indeed, it will also reflect that we are not limited to one ExoPlayer's non-fatal exception that we can catch.